### PR TITLE
Add tests for import assertions

### PR DESCRIPTION
**Pull request by jugglinmike**
"Import Assertions" is a proposal that is currently in stage 3 of TC39's standards process: https://github.com/tc39/proposal-import-assertions/

### CODE REVIEW

**Comment by jugglinmike** [comment]
@xtuc, @littledan, @MylesBorins, @dandclark

Hi there, "Import Assertions" proposal champions. I wrote these in collaboration with the V8 maintainers (cc @syg). If you have some time to review, then we can merge them to the `main` branch!

**Comment by rwaldron** [comment]
I only got through about 20 of them, but they look great so far.

File: test/language/module-code/import-assertion-many.js
Comment (dandclark) [inline test/language/module-code/import-assertion-many.js:5]: Looks like this didn't get updated after being copied from another test
```suggestion
    AssertClause in ImportDeclaration may contain multiple AssertEntries
```

**Review by dandclark — APPROVED** [review]
These looks great, thank you for these tests!
Just left a few minor comments.

File: src/dynamic-import/trailing-comma-second.case
Comment (jugglinmike) [reply src/dynamic-import/trailing-comma-second.case:14]: > Was the intent here was to have a trailing comma against the second param
> like this?

Yes, that was the intent. Thanks for pointing this out!

> Maybe it would be good to have a .case for both `import('', '')` and
> `import('', '',)` to validate that both are now syntactically valid.

Following the correction you spotted, I think we're good on from a coverage
standpoint. This patch's hand-written tests verify the former syntax, and
they're a little stronger than these procedurally-generated tests because they
also verify the semantics of those productions.

File: test/language/expressions/dynamic-import/2nd-param-assert-undefined.js
Comment (jugglinmike) [reply test/language/expressions/dynamic-import/2nd-param-assert-undefined.js:28]: My mistake. Removed.

File: test/language/module-code/import-assertion-many.js
Comment (jugglinmike) [reply test/language/module-code/import-assertion-many.js:5]: My mistake. Removed.

**Comment by jugglinmike** [comment]
Thanks for reviewing, @rwaldron and @dandclark!

**Comment by leobalter** [comment]
This set of tests includes the expansion of a simple dynamic import with a trailing comma. The tests capture that well, but test262 could benefit of a test for a full path where a dynamic import has a trailing comma, other than just checking syntax.

So in the cases you now have `import('', )` being a valid syntax, there should also be an assertion for the returned promise and paths for its resolution (resolved and rejected ends).

It's a very specific thing, so please take this suggestion as optional, @jugglinmike. Thanks for working on these tests!

**Comment by jugglinmike** [comment]
Heads up, reviewers: I've made another amendment based on some further
research. If a synchronous test includes an unhandled Promise rejection, the
test262-harness utility reports a failure in some engines. Even though this is
a little suspicious, I think the most pragmatic thing to do here is to switch
to using valid module specifiers even when they're not technically necessary.

Sorry for the churn; I'm hoping this isn't too controversial. I'll start a
separate discussion about whether Test262 needs a policy about that case.

> It's a very specific thing, so please take this suggestion as optional,
> @jugglinmike. Thanks for working on these tests!

Couldn't hurt. The patch now includes 4 tests for the resolution semantics of
ImportCall with trailing commas.

**Comment by leobalter** [comment]
> Couldn't hurt. The patch now includes 4 tests for the resolution semantics of
ImportCall with trailing commas.

I reviewed the test files from 21452eb and they LGTM.

**Comment by jugglinmike** [comment]
Thanks, @leobalter.

@dandclark In case you weren't aware: Test262 requires that pull requests receive approval from a TC39 organization member before they may be merged. Although in more colloquial contexts, I'd take your feedback to mean "this is ready to merge," I need at least one person to use GitHub's UI to formally "Approve" the pull request.

I don't mean to rush things, though! Just making sure you know about that more subtle requirement (it was news to me).

**Review by dandclark — APPROVED** [review]
The updates look good to me, thanks!

**Comment by jugglinmike** [comment]
Thanks again, @dandclark!

**Comment by jugglinmike** [comment]
Looks like I spoke too soon. Even though Dan's a member of the TC39 GitHub organization, their approval seems to be insufficient. I'm not an administrator of this repository, so I can't determine who *is* authorized to accept patches.

@rwaldron can you shed any light on this situation?

**Comment by rwaldron** [comment]
@jugglinmike Dan may be a member of TC39, but he's not a member of Test262. People you have write/admin access to this repo are those that have extensive, trusted contributions. For the last three years we've operated like this: 

1. Champions review the work and leave an approval (or not)
2. Test262 maintainer (with write/admin) does a follow up review and then merges.

**Review by rwaldron — DISMISSED** [review]

File: test/rejected.js
Comment (rwaldron) [inline test/rejected.js:1]: What does this file do? I don't think you meant to commit this?

**Review by rwaldron — CHANGES_REQUESTED** [review]

File: test/rejected.js
Comment (jugglinmike) [reply test/rejected.js]: That was research for gh-2988, and I didn't mean to commit it. Sorry about that; too many plates spinning. I've removed it.

Comment (rwaldron) [reply test/rejected.js]: No problem!

**Review by rwaldron — APPROVED** [review]

**Pull request merged**
